statistics/handle/globalstats/global_stats.go

```diff
@@ -80,188 +76,18 @@ func MergePartitionStats2GlobalStats(
-		} else {
-			okLoad = false
-		}
-		// If pre-load partition stats isn't provided, then we load partition stats directly and set it into allPartitionStats
-		if !okLoad {
-			var err1 error
-			partitionStats, err1 = loadTablePartitionStatsFn(tableInfo, &def)
-			if err1 != nil {
-				if skipMissingPartitionStats && types.ErrPartitionStatsMissing.Equal(err1) {
-					globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, fmt.Sprintf("partition `%s`", def.Name.L))
-					continue
-				}
-				err = err1
-				return
-			}
-			if externalCache {
-				allPartitionStats[partitionID] = partitionStats
-			}
-		}
-
-		for i := 0; i < globalStats.Num; i++ {
-			// GetStatsInfo will return the copy of the statsInfo, so we don't need to worry about the data race.
-			// partitionStats will be released after the for loop.
-			hg, cms, topN, fms, analyzed := partitionStats.GetStatsInfo(histIDs[i], isIndex, externalCache)
-			skipPartition := false
-			if !analyzed {
-				var missingPart string
-				if !isIndex {
-					missingPart = fmt.Sprintf("partition `%s` column `%s`", def.Name.L, tableInfo.FindColumnNameByID(histIDs[i]))
-				} else {
-					missingPart = fmt.Sprintf("partition `%s` index `%s`", def.Name.L, tableInfo.FindIndexNameByID(histIDs[i]))
-				}
-				if !skipMissingPartitionStats {
-					err = types.ErrPartitionStatsMissing.GenWithStackByArgs(fmt.Sprintf("table `%s` %s", tableInfo.Name.L, missingPart))
-					return
-				}
-				globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, missingPart)
-				skipPartition = true
-			}
-
-			// Partition stats is not empty but column stats(hist, topN) is missing.
-			if partitionStats.RealtimeCount > 0 && (hg == nil || hg.TotalRowCount() <= 0) && (topN == nil || topN.TotalCount() <= 0) {
-				var missingPart string
-				if !isIndex {
-					missingPart = fmt.Sprintf("partition `%s` column `%s`", def.Name.L, tableInfo.FindColumnNameByID(histIDs[i]))
-				} else {
-					missingPart = fmt.Sprintf("partition `%s` index `%s`", def.Name.L, tableInfo.FindIndexNameByID(histIDs[i]))
-				}
-				if !skipMissingPartitionStats {
-					err = types.ErrPartitionColumnStatsMissing.GenWithStackByArgs(fmt.Sprintf("table `%s` %s", tableInfo.Name.L, missingPart))
-					return
-				}
-				globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, missingPart+" hist and topN")
-				skipPartition = true
-			}
-
-			if i == 0 {
-				// In a partition, we will only update globalStats.Count once.
-				globalStats.Count += partitionStats.RealtimeCount
-				globalStats.ModifyCount += partitionStats.ModifyCount
-			}
-
-			if !skipPartition {
-				allHg[i] = append(allHg[i], hg)
-				allCms[i] = append(allCms[i], cms)
-				allTopN[i] = append(allTopN[i], topN)
-				allFms[i] = append(allFms[i], fms)
-			}
-		}
+	worker, err := NewAsyncMergePartitionStats2GlobalStats(gpool, globalTableInfo, histIDs, is, getTableByPhysicalIDFn, callWithSCtxFunc)
```

Comment:
Will it cost much effort to keep the old implementation, and separate the new implementation under a variable or fix control? Let's use the implementation for new cluster first.